# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/ipc/CanvasManagerChild.h
# Commit: ce28584e3869
# Full Hash: ce28584e386976fc6df815260cdfebd8e6ffbadd
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-19 05:06:09
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r bb4029ea69bb -r ce28584e3869 gfx/ipc/CanvasManagerChild.h
--- a/gfx/ipc/CanvasManagerChild.h	Mon Dec 18 18:10:46 2023 +0000
+++ b/gfx/ipc/CanvasManagerChild.h	Mon Dec 18 18:10:46 2023 +0000
@@ -49,6 +49,7 @@
   void EndCanvasTransaction();
   void ClearCachedResources();
   void DeactivateCanvas();
+  void BlockCanvas();
 
   RefPtr<layers::CanvasChild> GetCanvasChild();
 
@@ -63,6 +64,7 @@
   RefPtr<webgpu::WebGPUChild> mWebGPUChild;
   const uint32_t mId;
   bool mActive = true;
+  bool mBlocked = false;
 
   static MOZ_THREAD_LOCAL(CanvasManagerChild*) sLocalManager;
   static Atomic<uint32_t> sNextId;